### PR TITLE
added mobile tests in stubs

### PR DESCRIPTION
**Pull request by BiswajeetGope-Geek**
# PR Details

## Added tests for automation in mobile both Android and iOS

Please review the releases section of the repository to get the detailed information on changes.

## what it does?

* It tests automation for mobile in both `Android` and ` iOS` for native, browser as well as cloud level.
* It is based upon `appium` binding with `webdriverio`.
* It uses `xpaths`  based upon `android` and `ios`.
* It uses `chai` assertion library for checking the desired and expected results for test.
* Enables automatic `screenshots` whenever tests fails

## How Has This Been Tested

* It has been tested on `emulators`, `simulators` and `Android real devices` 

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

* Go over all the following points, and put an `x` in all the boxes that apply.
* If you're unsure about any of these, don't hesitate to ask. We're here to help!

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: questions/browser-stack/username.js [hunk 1]
```diff
@@ -3,6 +3,6 @@ module.exports = {
 	"type": "input",
 	"name": "browserStack.user",
 	"message": "Browser Stack Username?",
-	"when": (_answers) => _answers.arch === "web" && _answers.stack === "browser-stack",
+	"when": (_answers) => (_answers.arch === "web" && _answers.stack === "browser-stack") || (_answers.arch === "mobile" && _answers.stack === "browser-stack"),
```
Comment (faizahmedfarooqui) [inline questions/browser-stack/username.js]:
```js
"when": (_answers.arch == "web" || _answers.arch == "mobile") && _answers.stack === "browser-stack";
```

makes more sense to me...

File: questions/browser-stack/key.js [hunk 2]
```diff
@@ -3,6 +3,6 @@ module.exports = {
 	"type": "input",
 	"name": "browserStack.key",
 	"message": "Browser Stack Key?",
-	"when": (_answers) => _answers.arch === "web" && _answers.stack === "browser-stack",
+	"when": (_answers) => (_answers.arch === "web" && _answers.stack === "browser-stack") || (_answers.arch === "mobile" && _answers.stack === "browser-stack"),
```
Comment (faizahmedfarooqui) [inline questions/browser-stack/key.js]:
```js
"when": (_answers.arch == "web" || _answers.arch == "mobile") && _answers.stack === "browser-stack";
```

Makes more sense here...

File: questions/browser-stack/username.js (hunk 1 above)
Comment (BiswajeetGope-Geek) [reply questions/browser-stack/username.js]: yes

File: stubs/mobile/babel-config.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+let babelConfigJS = {};
+
+babelConfigJS.filename = 'babel.config.js';
+
+babelConfigJS.content = ` module.exports = {
+    presets: [
```
Comment (faizahmedfarooqui) [inline stubs/mobile/babel-config.js]: Indent using Tabs "2"

File: stubs/mobile/conf/android.browser.conf.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
+const { get } = require("lodash");
+const commonJs = require("./common.conf");
+
+let confJs = {}
+confJs.filename = "android.browser.conf.js";
+confJs.path = "conf";
+confJs.content = (answers) => {
+    const _commonPartial=commonJs.content(answers);
```
Comment (faizahmedfarooqui) [inline stubs/mobile/conf/android.browser.conf.js]: Indent using Tabs "2"

File: stubs/mobile/conf/android.conf.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
+const { get } = require("lodash");
+const commonJs = require("./common.conf");
+
```
Comment (faizahmedfarooqui) [inline stubs/mobile/conf/android.conf.js:3]: Indent using Tabs "2"

File: stubs/mobile/conf/browserstack-android.conf.js [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+const commonJs = require("./common.conf");
+
+let confJs = {}
+confJs.filename = "browserstack-android.conf.js";
+confJs.path = "conf";
+confJs.content = (answers) => {
+  const _commonPartial = commonJs.content(answers);
+  return `
+  exports.config =
+    {
```
Comment (faizahmedfarooqui) [inline stubs/mobile/conf/browserstack-android.conf.js]:
```js
exports.config = {
```

Comment (faizahmedfarooqui) [reply stubs/mobile/conf/browserstack-android.conf.js]: Indent using Tabs "2"

File: stubs/mobile/conf/common.conf.js [hunk 7]
```diff
@@ -0,0 +1,31 @@
+const {get}=require("loadash")
+let commonConfJs = {}
+commonConfJs.content = (answers) => {
+    return `
+    runner: 'local',
+        framework: 'mocha',
```
Comment (faizahmedfarooqui) [inline stubs/mobile/conf/common.conf.js]: Indent using Tabs "2"

File: stubs/mobile/conf/ios.browser.conf.js [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+const commonJs = require("./common.conf");
+
+let confJs = {}
+confJs.filename = "android.browser.conf.js";
+confJs.path = "conf";
+confJs.content = (answers) => {
+    const _commonPartial = commonJs.content(answers);
+    return `
+    exports.config = {
+        specs: [
```
Comment (faizahmedfarooqui) [inline stubs/mobile/conf/ios.browser.conf.js]: Indent using Tabs "2"

File: stubs/mobile/conf/ios.conf.js [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+const commonJs = require("./common.conf");
+
+let confJs = {}
+confJs.filename = "android.conf.js";
+confJs.path = "conf";
+confJs.content = (answers) => {
+    const _commonPartial = commonJs.content(answers);
+    return `
+ exports.config =
+        {
```
Comment (faizahmedfarooqui) [inline stubs/mobile/conf/ios.conf.js]: Indent using Tabs "2"

File: stubs/mobile/helpers/SelectClass.js [hunk 10]
```diff
@@ -0,0 +1,21 @@
+let selectClassJs = {};
+
+selectClassJs.path = "src/helpers";
+
+selectClassJs.filename = "SelectClass.js";
+
+selectClassJs.content =
+	`class SelectClass {
+        static _selectIos(xpath, text) {
```
Comment (faizahmedfarooqui) [inline stubs/mobile/helpers/SelectClass.js]: Indent using Tabs "2"

File: stubs/mobile/index.js [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+	testJs: require("./test"),
+
+	//conf ...
+	androidConfJs: require("./conf/android.conf"),
+	iosConfJs: require("./conf/ios.conf"),
+	browserStackAndroidConfJs: require("./conf/browserstack-android.conf"),
+	browserStackIosConfJs: require("./conf/browserstack-ios.conf"),
+	browserAndroidConfJs: require("./conf/android.browser.conf"),
+	browserIosConfJs: require("./conf/ios.browser.conf"),
+
```
Comment (faizahmedfarooqui) [inline stubs/mobile/index.js]: Indent using Tabs "2"

**Pull request merged**